BALSAMIC/assets/scripts/sex_prediction_tga.py

```diff
@@ -0,0 +1,487 @@
+import click
+import pandas as pd
+import os
+import warnings
+from typing import List, Optional, Dict, Any, Tuple
+
+from BALSAMIC.utils.io import write_json
+
+
+def read_cov(filepath: str) -> List[List[str]]:
+    """Read a coverage file and return its rows as a list of lists.
+
+    Args:
+        filepath (str): Path to the coverage file.
+
+    Returns:
+        List[List[str]]: Parsed rows of the coverage file.
+    """
+    with open(filepath, "r") as rf:
+        return [line.split("\t") for line in rf]
+
+
+def process_data(data_rows: List[List[str]]) -> Optional[pd.DataFrame]:
+    """Process coverage data rows to filter and structure into a DataFrame.
+
+    Args:
+        data_rows (List[List[str]]): Rows of coverage data.
+
+    Returns:
+        Optional[pd.DataFrame]: DataFrame with processed data or None if insufficient data.
+    """
+    filtered_data = [row for row in data_rows if row[0] in {"Y", "X"}]
+    if len(filtered_data) < 10:
+        warnings.warn(
+            "Cannot compute sex due to insufficient data, ignoring", UserWarning
+        )
+        return None
+
+    records: dict = {
+        f"{row[0]}_{row[1]}_{row[2]}": {"chrom": row[0], "depth": row[4]}
+        for row in filtered_data
+    }
+
+    df = pd.DataFrame.from_dict(records, orient="index")
+    df["depth"] = pd.to_numeric(df["depth"], errors="coerce")
+    return df
+
+
+def get_stats(df: pd.DataFrame) -> pd.DataFrame:
+    """Compute statistical summaries of depth grouped by chromosome.
+
+    Args:
+        df (pd.DataFrame): DataFrame containing coverage data.
+
+    Returns:
+        pd.DataFrame: DataFrame with statistics such as mean, median, min, max, etc.
+    """
+    return (
+        df.groupby("chrom")["depth"]
+        .agg(
+            mean="mean", median="median", min="min", max="max", std="std", count="count"
+        )
+        .reset_index()
+    )
+
+
+def extract_stats(stats: pd.DataFrame) -> Dict[str, float]:
+    """Extract statistics from a DataFrame into a dictionary.
+
+    Args:
+        stats (pd.DataFrame): DataFrame with statistical summaries.
+
+    Returns:
+        Dict[str, float]: Dictionary containing extracted statistics.
+    """
+
+    stats: dict = stats.T.to_dict()
+    sample_dict = {}
+    sample_dict["X_mean"] = stats[0]["mean"]
+    sample_dict["X_median"] = stats[0]["median"]
+    sample_dict["X_min"] = stats[0]["min"]
+    sample_dict["X_max"] = stats[0]["min"]
+    sample_dict["X_std"] = stats[0]["std"]
+    sample_dict["X_count"] = stats[0]["count"]
+    sample_dict["Y_mean"] = stats[1]["mean"]
+    sample_dict["Y_median"] = stats[1]["median"]
+    sample_dict["Y_min"] = stats[1]["min"]
+    sample_dict["Y_max"] = stats[1]["max"]
+    sample_dict["Y_std"] = stats[1]["std"]
+    sample_dict["Y_count"] = stats[1]["count"]
+    return sample_dict
```

Comment:
```suggestion
    stats_dict = stats.stack().to_dict()
    sample_dict = {
        f"{col}_{metric}": value
        for (col, metric), value in stats_dict.items()
    }
    
    return sample_dict
```
would this work instead?

Reply:
Done!